kolena/_experimental/object_detection/utils.py

```diff
@@ -16,12 +16,120 @@
 from typing import List
 from typing import Optional
 from typing import Set
+from typing import Tuple
 
+import numpy as np
+
+from kolena._extras.metrics.sklearn import sklearn_metrics
 from kolena._utils import log
 from kolena.workflow.evaluator import ConfusionMatrix
+from kolena.workflow.evaluator import Curve
+from kolena.workflow.evaluator import CurvePlot
 from kolena.workflow.metrics import MulticlassInferenceMatches
 
 
+def _compute_sklearn_arrays(
+    all_matches: List[MulticlassInferenceMatches],
+) -> Tuple[List[int], List[int], Dict[str, List[int]], Dict[str, List[int]]]:
+    y_true_by_label: defaultdict[str, List[int]] = defaultdict(lambda: [])
+    y_score_by_label: defaultdict[str, List[int]] = defaultdict(lambda: [])
+    y_true: List[int] = []
+    y_score: List[int] = []
+    for image_bbox_matches in all_matches:
+        for _, bbox_inf in image_bbox_matches.matched:  # TP (if above threshold)
+            y_true_by_label[bbox_inf.label].append(1)
+            y_score_by_label[bbox_inf.label].append(bbox_inf.score)
+            y_true.append(1)
+            y_score.append(bbox_inf.score)
+        for bbox_gt, _ in image_bbox_matches.unmatched_gt:  # FN
+            y_true.append(1)
+            y_score.append(-1)
+            y_true_by_label[bbox_gt.label].append(1)
+            y_score_by_label[bbox_gt.label].append(-1)
```

Comment:
This causes strange plots for all-negative test cases. Is there a reason to prefer `-1` over `0`?

Reply:
Update on this, I'm keeping it to be -1 to force `unmatched_gt`s to always be under the threshold, even if the threshold is 0. When the threshold is 0, and I set an `unmatched_gt`'s `y_score` to be 0 and filter with `>=`, then `y_true` is 1 and `y_pred` is 1, which is a TP.
What's different now is that every threshold is guaranteed to be non-negative, so no `-1`s exist in the x-axis anymore.